backend/src/schema/resolvers/groups.js

```diff
@@ -10,30 +10,42 @@ import { mergeImage } from './images/images'
 export default {
   Query: {
     Group: async (_object, params, context, _resolveInfo) => {
-      const { id: groupId, isMember } = params
+      const { isMember, id, slug } = params
+      const matchParams = { id, slug }
+      Object.keys(matchParams).forEach((key) => {
+        if ([undefined, null].includes(matchParams[key])) {
+          delete matchParams[key]
+        }
+      })
       const session = context.driver.session()
       const readTxResultPromise = session.readTransaction(async (txc) => {
-        const groupIdCypher = groupId ? ` {id: "${groupId}"}` : ''
+        const matchParamsEntries = Object.entries(matchParams)
+        let groupMatchParamsCypher = ''
+        matchParamsEntries.forEach((ele, index) => {
+          groupMatchParamsCypher += index === 0 ? ' {' : ''
+          groupMatchParamsCypher += `${ele[0]}: "${ele[1]}"`
+          groupMatchParamsCypher += index < matchParamsEntries.length - 1 ? ', ' : '}'
+        })
```

Comment:
Can this be done in some human readable way? Am I right, that we receive an ID, a slug, both or none from the params. So I suggest, if both (ID and slug) are undefined, return empty array or throw an error. We could say, that ID goes over slug if both are present.

Reply:
I couldn't find a good solution in Cypher.
What I have found was not working. So I spent only an hour. @Mogge 

I created two hopefully well named and therefore descriptional helper functions instead. Hope that satisfies you …